src/sdam/description/topology/test/sdam.rs

```diff
@@ -444,6 +510,87 @@ async fn errors() {
     run_spec_test(&["server-discovery-and-monitoring", "errors"], run_test).await;
 }
 
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn monitoring() {
+    run_spec_test(&["server-discovery-and-monitoring", "monitoring"], run_test).await;
+}
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+#[function_name::named]
+async fn topology_closed_event_last() {
+    let _guard: RwLockReadGuard<_> = LOCK.run_concurrently().await;
+
+    let client = EventClient::new().await;
+    client
+        .database(function_name!())
+        .collection(function_name!())
+        .insert_one(doc! { "x": 1 }, None)
+        .await
+        .unwrap();
+
+    let handler = client.handler.clone();
+    drop(client);
+
+    let last_event = handler.get_all_sdam_events().into_iter().last().unwrap();
```

Comment:
I think it would be better to use the channel to receive the events here to avoid racing the `drop` implementation of `Topology`. To achieve the "last" assertion, you can just use a filter that matches any event and assert the timeout is reached.